conda/env/installers/conda.py

```diff
@@ -56,18 +57,18 @@ def _solve(
 
 def dry_run(
     specs: list[str], args: Namespace, env: Environment, *_, **kwargs
-) -> Environment:
+) -> EnvironmentYaml:
```

Comment:
We want to continue to return the `conda.env.env.Environment` here so that we aren't changing the installer api. Eventually this should move over to use the new Environment model. But I think it's better to chunk that in another piece of work to keep this change focused on the env spec api.